pandas/tests/base/test_value_counts.py

```diff
@@ -274,3 +274,13 @@ def test_value_counts_datetime64(index_or_series):
     td2 = klass(td2, name="dt")
     result2 = td2.value_counts()
     tm.assert_series_equal(result2, expected_s)
+
+
+def test_value_counts_with_nan(index_or_series):
+    # GH31944
+    klass = index_or_series
+    values = [True, pd.NA, np.nan]
+    s = klass(values)
+    expected = Series([2, 1], index=[pd.NA, True])
```

Comment:
nit-pick: can you move this to after res
That way this reads: construct result, construct expected result, compare result with expected

Reply:
Sure, makes sense!